octobot/community/history_backend/iceberg_historical_backend_client.py

```diff
@@ -0,0 +1,310 @@
+#  This file is part of OctoBot (https://github.com/Drakkar-Software/OctoBot)
+#  Copyright (c) 2025 Drakkar-Software, All rights reserved.
+#
+#  OctoBot is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  OctoBot is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public
+#  License along with OctoBot. If not, see <https://www.gnu.org/licenses/>.
+import typing
+import enum
+import datetime
+import os
+import asyncio
+import concurrent.futures
+
+import pyarrow
+import pyiceberg.catalog
+import pyiceberg.schema
+import pyiceberg.types
+import pyiceberg.exceptions
+import pyiceberg.expressions
+import pyiceberg.table
+import pyiceberg.table.sorting
+
+import octobot_commons.logging as commons_logging
+import octobot_commons.enums as commons_enums
+import octobot.constants as constants
+import octobot.community.history_backend.historical_backend_client as historical_backend_client
+import octobot.community.history_backend.util as history_backend_util
+
+
+class TableNames(enum.Enum):
+    OHLCV_HISTORY = constants.ICEBERG_OHLCV_HISTORY_TABLE
+
+
+# pyiceberg itself uses a thread pool so each thread will create children threads, limit the number of "master" threads
+_MAX_EXECUTOR_WORKERS = min(4, (os.cpu_count() or 1)) # use a max of 4 workers
+
+
+class IcebergHistoricalBackendClient(historical_backend_client.HistoricalBackendClient):
+
+    def __init__(self):
+        self.namespace: typing.Optional[str] = None
+        self.catalog: pyiceberg.catalog.Catalog = None # type: ignore
+        self._executor: typing.Optional[concurrent.futures.ThreadPoolExecutor] = None # only availble when client is open
+        
+    async def open(self):
+        try:
+            self._executor = concurrent.futures.ThreadPoolExecutor(max_workers=_MAX_EXECUTOR_WORKERS)
```

Comment:
I added this to use a local executor that is limited and closed when we don't need it anymore